plugin/storage/kafka/unmarshaller.go

```diff
@@ -57,3 +60,34 @@ func (h *JSONUnmarshaller) Unmarshal(msg []byte) (*model.Span, error) {
 	err := jsonpb.Unmarshal(bytes.NewReader(msg), newSpan)
 	return newSpan, err
 }
+
+// zipkinThriftUnmarshaller implements Unmarshaller
+type zipkinThriftUnmarshaller struct{}
+
+// NewZipkinThriftUnmarshaller constructs a zipkinThriftUnmarshaller
+func NewZipkinThriftUnmarshaller() *zipkinThriftUnmarshaller {
+	return &zipkinThriftUnmarshaller{}
+}
+
+// Unmarshal decodes a json byte array to a span
+func (h *zipkinThriftUnmarshaller) Unmarshal(msg []byte) (*model.Span, error) {
+	buffer := thrift.NewTMemoryBuffer()
+	buffer.Write(msg)
+
+	transport := thrift.NewTBinaryProtocolTransport(buffer)
+	_, _, err := transport.ReadListBegin() // Ignore the returned element type
+	if err != nil {
+		return nil, err
+	}
+
+	zs := &zipkincore.Span{}
+	if err = zs.Read(transport); err != nil {
+		return nil, err
+	}
+	mSpan, err := zipkin.ToDomainSpan(zs)
+
+	if err != nil {
+		return nil, err
+	}
+	return mSpan[0], err
```

Comment:
@yurishkuro I am not sure it's some good code, any advices to avoid using [0]. I thought about changing the interface to return []model.Span and handle the array in the calling function.

Reply:
I think the ingester is currently built with the assumption that a single Kafka message contains just a single span. Even in this case a single Zipkin span can sometimes be transformed to two Jaeger spans, so it would be better to extend the ingester to support the notion of many spans per message, at least after unmarshaling. However, I would not recommend doing it in this PR, it should be a separate PR as it might get quite large. For now I would suggest logging a warning.

There are good reasons to expect a single span per Kafka message, it makes the streaming jobs easier to write.

Reply:
Thanks